### ISSUE TITLE
Add support for Dockershim

### ISSUE THREAD

**Issue by Jono-SUSE-Rancher** [rancher/rancher#31943]
**Dockershim**
As of k8s 1.20, Dockershim, the Container Runtime Interface (CRI) compliant layer between the kubelet and the Docker Daemon will no longer be supported.

**What We Need To Do**
We will implement the upstream open source community Dockershim announced by Mirantis and Docker to ensure RKE clusters can continue to leverage Docker as their container runtime.

**For More Information**
For more information, please see the official [Rancher Announcement](https://support.rancher.com/hc/en-us/articles/360053308831-Rancher-Operational-Advisory-Related-to-deprecation-of-dockershim-in-Kubernetes-v1-20#what-is-dockershim--0-0) in regards to Dockershim.

**Comment by superseb** [comment]
To enable (only for k8s v1.21 and higher):

```
enable_cri_dockerd: true
```

**Comment by izaac** [comment]
@superseb could we have more information on the testing scope for this? And how do we validate we have this enabled in the nodes?

**Comment by superseb** [comment]
* Should only be possible to enable on k8s v1.21 and higher
* When enabled, kubelet should be configured to use the new container runtime endpoint:
```
$ docker inspect kubelet | egrep "container-runtime" | sed 's/ //g'| sort -u
"--container-runtime-endpoint=/var/run/dockershim.sock",
"--container-runtime=remote",
```
* When enabled, `cri-dockerd` should be running on the hosts
```
# ps -ef | grep cri-dockerd | grep -v grep
root     13584 13164  0 13:29 ?        00:00:06 /opt/rke-tools/bin/cri-dockerd
```
* All tests should complete successfully

* When `cri-dockerd` process is manually killed on the host, `kubelet` container should exit and automatically restart.
* When `kubelet` process is manually killed on the host, `kubelet` container should exit and automatically restart.


When not enabled, there should be no difference in any behavior, no extra parameters should be configured on `kubelet` and no `cri-dockerd` process should be running.

**Comment by izaac** [comment]
Rancher version: `2.6 master-head (06/23/2021)` _d44698d_
On an initial validation this option is working while used with RKE, but the Rancher Edit as YAML is not persisting this option as expected.  

Opened a separate issue for this.
#33358

I will do the validations with RKE provisioned clusters as [previously mentioned](https://github.com/rancher/rancher/issues/31943#issuecomment-859593726).

**Comment by izaac** [comment]
RKE `1.3.0-rc7`
`kubernetes_version: v1.21.1-rancher2-1`

OL = Oracle Linux

OL7.9 SELinux 
Docker 19.03.15
rancher-selinux-0.2-rc2
**Pass**


OL8.3 SELinux 
Docker 20.10.7
rancher-selinux-0.2-rc2
**Pass**


Ubuntu 20.04
Docker 19.03.15
**Pass**


Ubuntu 18.04
Docker 20.10.7
**Pass**

CentOS 7.9 SELinux
Docker 20.10.7
rancher-selinux-0.2-rc2
**Pass**

CentOS 8.3 SELinux 
Docker 19.03.15
rancher-selinux-0.2-rc2
**Pass**


I will do a sanity check on Ubuntu, OL, and CentOS when [this yaml fix](https://github.com/rancher/rancher/issues/33358) is in.

**Comment by izaac** [comment]
Blocked by: https://github.com/rancher/rancher/issues/33445

**Comment by izaac** [comment]
Rancher version `2.6-head (08/02/2021)` _6c14c33_

Remaining test execution was done as part of this issue:
https://github.com/rancher/rancher/issues/33445#issuecomment-892201144

Including a sanity check on Ubuntu 20.04
